### PR TITLE
Make aie2xclbin stop when any intermediate step fails 

### PR DESCRIPTION
**Pull request by newling**
Fix for bug introduced in https://github.com/nod-ai/iree-amd-aie/commit/7d530aff2146bc5d73e6b051d65f45757aa6c3b2 where aie2xclbin keeps running even after a tool fails. Tested locally.

### CODE REVIEW

**Comment by makslevental** [comment]
> The weird and worrying thing is that the e2e numerical tests passed for me locally when I gave an invalid path to peano -- nothing was being compiled yet the numerical test passed

Were you manually setting tempdir? If so, it's because the workdir isn't cleared and somewhere (haven't found it yet) there's some optimization that reuses existing (ie stale from a previous successful run, in your case) object files.

**Comment by newling** [comment]
> > The weird and worrying thing is that the e2e numerical tests passed for me locally when I gave an invalid path to peano -- nothing was being compiled yet the numerical test passed
> 
> Were you manually setting tempdir? If so, it's because the workdir isn't cleared and somewhere (haven't found it yet) there's some optimization that reuses existing (ie stale from a previous successful run, in your case) object files.

Yes, that must be it 🧠

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp [hunk 1]
```diff
@@ -187,10 +187,20 @@ Expected<std::string> runTool(
     llvm::outs() << buffer.str();
   }
 
+  if (result != 0) {
+    return make_error<StringError>(
```
Comment (makslevental) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp:191]: might as well move this to right after the `sys::ExecuteAndWait`? doesn't really matter

**Review by makslevental — APPROVED** [review]

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp (hunk 1 above)
Comment (newling) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp]: Yeah maybe printing whatever stdouts captured first will be useful though

**Pull request merged**